app/helpers/security_helper.rb

```diff
@@ -19,13 +19,25 @@ def add_stream_cookies(stream_info)
     end
   end
 
-  def secure_streams(stream_info)
+  def secure_streams(stream_info, media_object_id)
+    begin
+      if Avalon::Configuration.controlled_digital_lending_enabled? && Checkout.checked_out_to_user(media_object_id, current_user.id).empty?
+        raise StreamToken::Unauthorized
+      else
+        add_stream_url(stream_info)
+      end
```

Comment:
I wonder if this could be rewritten as the following and work.
```suggestion
        add_stream_url(stream_info) if Avalon::Configuration.controlled_digital_lending_enabled? && Checkout.checked_out_to_user(media_object_id, current_user&.id).empty?
```
Or if that doesn't work then adding another clause:

```suggestion
        add_stream_url(stream_info) if Avalon::Configuration.controlled_digital_lending_enabled? && current_user && Checkout.checked_out_to_user(media_object_id, current_user.id).empty?
```
At that point a small private method to contain the conditional logic might make sense.